### DISCUSSION TITLE
Triggering simulations from outside OMNET++ and dynamic interaction with external components

### DISCUSSION THREAD

**Discussion by Abh4git** [omnetpp/omnetpp#993] [Q&A] [answered]
My name is Abhilash and I am a PhD scholar working in the area of Time Sensitive Networks. I am implementing a scheduling approach for ethernet and want to pass this on to OMNET++ and trigger simulation. Can I do this? If so can you please provide me some pointers on the same.

Also I would like to take also the simulation results as it goes on and pass on to my external components. Is this possible as well? 
Any pointers would be helpful.

From my understanding, as a novice, noticed that OMNET++ compilation also includes the topology ned file and therefore I was kind of feeling only limited integration might be possible. But maybe I am wrong and someone has figured it out.

Thanks a lot for the inputs!

**Comment by rhornig** [comment, marked as answer]
If you intend to work with TSN networks, you should take a look at the [TSN framework in INET 4.5](https://inet.omnetpp.org/docs/showcases/tsn). You can implement your own scheduling mechanism in C++. Interfacing with external components are totally possible as OMNeT++ is just a C++ library at the end, so anything you can do in C++ can be done in the simulator. NED files are just convenience mechanisms to set up your network topologies, but behaviour is always defined in C++ code.

**Reply by Abh4git** [reply]
Thank you very much! Helps a lot in taking my next steps!
